Better detection of TV devices

B0pol opened the pull request:
#### What is it?
- [x] Bug fix (user facing)
- [ ] Feature (user facing)
- [ ] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
<!-- While bullet points are the norm in this section, feel free to write a text instead if you can't fit it in a list -->
- have a better detection of TV devices, so that everyone with a TV can have a great experience on NewPipe.

Because some boxes — including mine — are just boxes shipped with Android with a TV interface / remote-friendly interface, but they aren't shipped with [_official_ Android TV system](https://www.android.com/tv/), i.e. with Google Assistant, Chromecast… , or Fire TV (I've seen Fire TV detection in `AndroidTvUtils.isTv()`).

Please include this PR in 0.19.3, this way https://github.com/TeamNewPipe/NewPipe/pull/3478 will ship all TV fixes.

#### Fixes the following issue(s)
<!-- Also add reddit or other links which are relevant to your change. -->
- Some android boxes aren't detected as TV (because not Fire TV / Android TV), and it created many issues mainly related with navigation.

#### Testing apk
<!-- Ensure that you have your changes on a new branch which has a meaningful name. This name will be used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe. Do NOT name your branches like "patch-0" and "feature-1". For example, if your PR implements a bug fix for comments, an appropriate branch name would be "commentfix". -->
[NewPipe_tv_detection.zip](https://github.com/TeamNewPipe/NewPipe/files/4532762/NewPipe_tv_detection.zip)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

TobiGr commented:
On KitKat
``` log
org.schabi.newpipe.debug.tv E/class org.schabi.newpipe.report.ErrorActivity: java.lang.NoSuchMethodError: android.os.BatteryManager.getIntProperty
        at org.schabi.newpipe.util.AndroidTvUtils.isTv(AndroidTvUtils.java:32)
        at org.schabi.newpipe.MainActivity.onCreate(MainActivity.java:148)
        at android.app.Activity.performCreate(Activity.java:5231)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
        at android.app.ActivityThread.access$800(ActivityThread.java:135)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:5017)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:515)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
        at dalvik.system.NativeStart.main(Native Method)
```

B0pol commented:
I removed the suppress lint annotation and correctly handled sdk version. I've installed an android 4.4 emulator (it's super fast compared to android 10 emulator) and fixed the crash @TobiGr.

TobiGr commented:
we should store `isTV` as static class var to prevent calculating it again and again since this is more complex now. after that it should be good to go.

wb9688 commented on app/src/main/java/org/schabi/newpipe/util/AndroidTvUtils.java:
```diff
@@ -1,22 +1,55 @@
 public final class AndroidTvUtils {
+
     private static final String AMAZON_FEATURE_FIRE_TV = "amazon.hardware.fire_tv";
+    private static Boolean isTV = null;
```
Please use the primitive type `boolean`

B0pol replied:
No??
How do you know if it's the first time you check `isTv()` and have to do all the calculations? For the first time, object value is null, but primitive would be false, how do you difference unitialized `false` from `false` _real_ value with primitive?

I don't see any other proper way to acheive this https://github.com/TeamNewPipe/NewPipe/pull/3483#issuecomment-619447980 than using Boolean object and its null value, but you tell me.

wb9688 replied:
Oh, you're right, I forgot about that
